spec-0007/index.md

```diff
@@ -97,44 +97,69 @@ The [deprecation strategy](https://github.com/scientific-python/specs/pull/180#i
 
    ```
 
-   Also note the suggested language for the `rng` parameter docstring.
+   Also note the suggested language for the `rng` parameter docstring, which encourages the user to pass a `Generator` or None, but allows for other types accepted by `numpy.random.default_rng`.
 
 ### Impact
 
-The following users will be affected:
-
-1. Those who use `np.random.seed`.
-   The proposal will do away with that global seeding mechanism, meaning that code that relies on it will, after a certain deprecation period, start seeing a different stream of random numbers than before.
-   To ensure that this does not go unnoticed, the library should raise a `FutureWarning` if `np.random.seed` was called earlier (we show how to do that further down).
+There are three classes of users, which will be affected to varying degrees.
 
-   Code will, in effect, go from being seeded to being unseeded. To avoid that from happening, users should switch from using `np.random.seed` to specifying the `rng` argument to all functions that allow it.
+1. Those who do not attempt to control the random state.
+   Their code will immediately switch from using the unseeded global `RandomState` to using an unseeded `Generator`.
+   Since the underlying _distributions_ of pseudo-random numbers will not change, these users will be relatively unaffected.
 
-2. Those who do not seed. Their code will, after the deprecation period, use the newly proposed default. Since they were already not requesting repeatable sequences, and since the underlying _distributions_ of pseudo-random numbers did not change, they should be unaffected.
+2. Users of `random_state`/`seed` arguments. 
+   Support for these arguments will be dropped eventually, but during the deprecation period, we can provide clear guidance, via warnings and documentation, on how to migrate to the new `rng` keyword.
 
-3. Users of `random_state=...`.
-   Support for the `random_state` argument may be dropped eventually, at which point using that keyword will raise an error.
-   Meanwhile we can provide clear guidance, via deprecation warnings and documentation, on how to migrate to the new `rng` keyword.
+3. Those who use `numpy.random.seed`. 
+   The proposal will do away with that global seeding mechanism, meaning that code that relies on it will, after the deprecation period, go from being seeded to being unseeded.
+   To ensure that this does not go unnoticed, libraries that allowed for control of the random state via `numpy.random.seed` should raise a `FutureWarning` if `np.random.seed` has been called. (See Code below for an example.)
```

Comment:
```suggestion
   To ensure that this does not go unnoticed, libraries that allowed for control of the random state via `numpy.random.seed` should raise a `FutureWarning` when using the global random state after `np.random.seed` has been called. (See Code below for an example of how to check.)
```

Reply:
I'm not sure this is right; isn't the intention to warn in two cases: when using `rng` along with global seeding, or when using the global seed when `random_state=None`? So, I think any time that we detect that `np.random.seed` has been called, we need to warn the user that they're doing something wrong.

Reply:
> when using rng along with global seeding

How can `rng` be used with global seeding? Arguments to `rng` are validated by `default_rng`, and that will never return the global `RandomState` (or any `RandomState`).

> when using the global seed when `random_state=None`

Yes, currently when `random_state=None` *because it has not been specified* and `np.random.seed` has been called, we give a `FutureWarning` that the global random state won't work in the future.

When `random_state` *is specified explicitly*, whether it is `None`, `np.random`, or something else, we need to give a warning about `random_state` being used. If it was used by keyword, that is deprecated. It it was used positionally, there's a `FutureWarning` about that behavior changing. Either way, the user needs to use `rng` with a valid input to avoid the warning, and `rng` doesn't permit use of the global random state, so I don't think we *also" need to give a warning against use of the global random state. But we can discuss it.

Reply:
I meant by that: calling seed and then also specifying an rng. Sorry for being unclear.

Reply:
Perhaps something like: libraries should raise a FutureWarning when the `random_state` argument is set in a way that accesses global random state (this will not longer be supported via `rng`).

Reply:
Feel free to change it. To avoid more merge conflicts, I will not be working on this PR locally.